GoodFirstIssue-ToDoApp/Screens/ItemEdit/ItemEditViewController.swift

```diff
@@ -8,12 +8,43 @@
 
 import UIKit
 
-class ItemEditViewController: UIViewController {
+class ItemEditViewController: UIViewController, UITextFieldDelegate {
 
+    @IBOutlet weak var cancelButton: UIBarButtonItem!
+    @IBOutlet weak var saveButton: UIBarButtonItem!
+    @IBOutlet weak var nameTextField: UITextField!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         title = "項目追加"
+        nameTextField.delegate = self
+    }
+    
+    @IBAction func tapCancel(_ sender: Any) {
+        dismiss(animated: false)
     }
     
+    @IBAction func tapSave(_ sender: Any) {
+        dismiss(animated: false)
+    }
+    
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        if nameTextField.text != "" {
+            saveButton.isEnabled = true
+        }
+    }
     
+    func textFieldDidChangeSelection(_ textField: UITextField) {
+        if nameTextField.text != "" {
```

Comment:
編集中にもtextが空になった場合はfalseにする必要があるので、ifブロックを使う代わりに、以下のようにするのが良さそうです。

```suggestion
        saveButton.isEnabled = nameTextField.text != ""
```

Reply:
こんな書き方が出来るなんて思いつかなかったです。早速書き換えました。

Reply:
僕もこの書き方初めて知りました。勉強になります！